flink-libraries/flink-table/src/test/scala/org/apache/flink/table/api/scala/stream/sql/SqlITCase.scala

```diff
@@ -317,4 +317,92 @@ class SqlITCase extends StreamingWithStateTestBase {
     result.addSink(new StreamITCase.StringSink)
     env.execute()
   }
+  
+  
+  @Test
+  def testAvgSumAggregatationPartition(): Unit = {
+
+    val env = StreamExecutionEnvironment.getExecutionEnvironment
+    val tEnv = TableEnvironment.getTableEnvironment(env)
+    env.setParallelism(1)
+    StreamITCase.testResults = mutable.MutableList()
+
+    val sqlQuery = "SELECT a, AVG(c) OVER (PARTITION BY a ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) AS avgC," + 
+      "SUM(c) OVER (PARTITION BY a ORDER BY procTime()" +
+      "RANGE BETWEEN INTERVAL '10' SECOND PRECEDING AND CURRENT ROW) as sumC FROM MyTable"
+
+    val t = StreamTestData.get5TupleDataStream(env)
+         .assignTimestampsAndWatermarks(new ProcTimeTimestamp())
```

Comment:
@fhueske - I agree but Flink as it operates now does not have the timestamp set to the processing time (basically if you check the timestamp of record it is not set if you do not have this set). Therefore you have 2 possibilities: one: to move the assignment of time in the SQL implementation (as it was initially) or to keep it here and expect this is set.
before i had this logic in the processing logic - but your remark was that this should not be set there. Therefore now i moved it here. 
I do not see any alternative as this is quite a nasty case where we do not operate just on time nor just on number of elements.